x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -369,6 +394,38 @@ export const reindexServiceFactory = (
   // ------ The service itself
 
   return {
+    async hasRequiredPrivileges(indexName: string) {
+      const body = {
+        cluster: ['manage'],
+        index: [
+          {
+            names: [`${indexName}*`],
+            privileges: ['all'],
+          },
+          {
+            names: ['.tasks'],
+            privileges: ['read', 'delete'],
+          },
+        ],
+      } as any;
+
+      if (isMlIndex(indexName)) {
+        body.cluster = [...body.cluster, 'manage_ml'];
+      }
+
+      if (isWatcherIndex(indexName)) {
+        body.cluster = [...body.cluster, 'manage_watcher'];
+      }
+
+      const resp = await callCluster('transport.request', {
```

Comment:
Will need to catch the error thrown when security is not available:

```
server   error  [16:46:07.469]  [exception] Security must be explicitly enabled when using a trial license. Enable security by setting [xpack.security.enabled] to [true] in the elasticsearch.yml file and restart the node. :: {"path":"/_xpack/security/user/_has_privileges","body":"{\"cluster\":[\"manage\"],\"index\":[{\"names\":[\".ml-notifications*\"],\"privileges\":[\"all\"]},{\"names\":[\".tasks\"],\"privileges\":[\"read\",\"delete\"]}]}","statusCode":500,"response":"{\"error\":{\"root_cause\":[{\"type\":\"exception\",\"reason\":\"Security must be explicitly enabled when using a trial license. Enable security by setting [xpack.security.enabled] to [true] in the elasticsearch.yml file and restart the node.\"}],\"type\":\"exception\",\"reason\":\"Security must be explicitly enabled when using a trial license. Enable security by setting [xpack.security.enabled] to [true] in the elasticsearch.yml file and restart the node.\"},\"status\":500}"}
    at respond (/mnt/c/elastic/kibana/node_modules/elasticsearch/src/lib/transport.js:308:15)
    at checkRespForFailure (/mnt/c/elastic/kibana/node_modules/elasticsearch/src/lib/transport.js:267:7)
    at HttpConnector.<anonymous> (/mnt/c/elastic/kibana/node_modules/elasticsearch/src/lib/connectors/http.js:165:7)
    at IncomingMessage.wrapper (/mnt/c/elastic/kibana/node_modules/elasticsearch/node_modules/lodash/lodash.js:4935:19)
    at IncomingMessage.emit (events.js:187:15)
    at endReadableNT (_stream_readable.js:1094:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

Reply:
Pushed a fix that checks that security is enabled first.